### PR TITLE
[1.21.3] Bump CoreMods to 5.2

### PR DESCRIPTION
**Pull request by Jonathing**
This PR bumps CoreMods to 5.2.

Full Changelog:
https://gist.github.com/Jonathing/c3ad28b2a048ac839a7baba5417ee870

The key features are:
- ES6 language support
- Thoroughly updated ASMAPI, with full documentation
- Bug fixes (some optional for backwards-compatibility)
- Partial internal code cleanup

### CODE REVIEW

**Comment by Jonathing** [comment]
Once CoreMods 5.2.0 has been merged and this PR fully tested, I will mark it as ready for review.

File: src/main/resources/coremods/method_redirector.js [hunk 1]
```diff
@@ -1,77 +1,80 @@
...
-        false);
-}
-
-function contains(list, target) {
-    for(var i = 0; i < list.length; i++) {
-        if(list[i] == target) return true;
-    }
-    return false;
-}
+'use strict'; // prevents us from making sloppy mistakes
```
Comment (PaintNinja) [inline src/main/resources/coremods/method_redirector.js]: Redundant code comment

File: src/main/resources/coremods/method_redirector.js [hunk 2]
```diff
@@ -1,77 +1,80 @@
...
-function search(className, node, replacements) {
-    for(var i = 0; i < replacements.length; i++){
-        var r = replacements[i];
-        if(contains(r.targets, className) && node.getOpcode() == r.opcode && node.name == r.name && node.desc == r.desc) {
-            return r;
-        }
-    }
-    return null;
-}
+var ASMAPI = Java.type('net.minecraftforge.coremod.api.ASMAPI');
```
Comment (PaintNinja) [inline src/main/resources/coremods/method_redirector.js]: Can this be `const`?

**Comment by Jonathing** [comment]
I've added logging and an error case (just prints, doesn't crash the game) a class that should have method redirects does not have any applied. This will help alleviate future problems when porting the game, and addresses #10157. I will fix it in a future PR.

**Comment by Jonathing** [comment]
I have moved the work on cleaning up Forge's CoreMods to #10158. This PR now only contains the bump to CoreMods 5.2.

**Pull request merged**